src/index_notation/transformations.cpp

```diff
@@ -1487,6 +1492,310 @@ ir::Stmt GEMM::replaceValidStmt(IndexStmt stmt,
+    unconst_cast_stream << "const_cast<" << aTensorVar.getType().getDataType() << "*>";
+    unconst_cast = std::move(unconst_cast_stream).str();
+  }
+
+  // Construct shapes for TBLIS.
+  auto aShape = ir::Var::make("aShape", Auto);
+  auto bShape = ir::Var::make("bShape", Auto);
+  auto cShape = ir::Var::make("cShape", Auto);
+
+  results.push_back(ir::VarDecl::make(
+      aShape,
+      ir::Call::make(
+          "std::vector<tblis::len_type>", {aTensorVar.getOrder()}, Auto)));
+  for (int i = 0; i < aTensorVar.getOrder(); ++i) {
+    results.push_back(ir::Store::make(aShape, i, getLen(aDomain, i)));
+  }
+
+  results.push_back(ir::VarDecl::make(
+      bShape,
+      ir::Call::make(
+          "std::vector<tblis::len_type>", {bTensorVar.getOrder()}, Auto)));
+  for (int i = 0; i < bTensorVar.getOrder(); ++i) {
+    results.push_back(ir::Store::make(bShape, i, getLen(bDomain, i)));
+  }
+
+  results.push_back(ir::VarDecl::make(
+      cShape,
+      ir::Call::make(
+          "std::vector<tblis::len_type>", {cTensorVar.getOrder()}, Auto)));
+  for (int i = 0; i < cTensorVar.getOrder(); ++i) {
+    results.push_back(ir::Store::make(cShape, i, getLen(cDomain, i)));
+  }
+
+  // Construct strides for TBLIS.
+  auto aStrides = ir::Var::make("aStrides", Auto);
+  auto bStrides = ir::Var::make("bStrides", Auto);
+  auto cStrides = ir::Var::make("cStrides", Auto);
+
+  results.push_back(ir::VarDecl::make(
+      aStrides,
+      ir::Call::make(
+          "std::vector<tblis::stride_type>", {aTensorVar.getOrder()}, Auto)));
+  for (int i = 0; i < aTensorVar.getOrder(); ++i) {
+    results.push_back(ir::Store::make(aStrides, i, getStride(aAccess, i)));
+  }
+
+  results.push_back(ir::VarDecl::make(
+      bStrides,
+      ir::Call::make(
+          "std::vector<tblis::stride_type>", {bTensorVar.getOrder()}, Auto)));
+  for (int i = 0; i < bTensorVar.getOrder(); ++i) {
+    results.push_back(ir::Store::make(bStrides, i, getStride(bAccess, i)));
+  }
+
+  results.push_back(ir::VarDecl::make(
+      cStrides,
+      ir::Call::make(
+          "std::vector<tblis::stride_type>", {cTensorVar.getOrder()}, Auto)));
+  for (int i = 0; i < cTensorVar.getOrder(); ++i) {
+    results.push_back(ir::Store::make(cStrides, i, getStride(cAccess, i)));
+  }
+
+  // Create TBLIS tensors.
+  Datatype TblisTensor("tblis::tblis_tensor");
+  auto aTensor = ir::Var::make("aTensor", TblisTensor);
+  auto bTensor = ir::Var::make("bTensor", TblisTensor);
+  auto cTensor = ir::Var::make("cTensor", TblisTensor);
+
+  std::vector<ir::Expr> aArgs = {
+    ir::Call::make("&", {aTensor}, Auto),
+    aTensorVar.getOrder(), // ndim
+    ir::MethodCall::make(aShape, "data", {}, false /* deref */, Auto), // shape
+    getPtr(aAccess, aDomain), // data
+    ir::MethodCall::make(aStrides, "data", {}, false /* deref */, Auto), // strides
+  };
+  results.push_back(ir::VarDecl::make(aTensor, ir::Symbol::make("{}")));
+  // TODO (rohany): Pick the right call between double and float here.
```

Comment:
You can probably address this TODO here by doing something similar to https://github.com/rohany/taco/pull/118/files#diff-740b3a2fcbef09458f763dba3d5e2d2c9484b21a3bc88b58b9cf5d4ab34083f6R1480